### ISSUE TITLE
Getting "Error: Parse Error: Invalid header value char" In Postman

### ISSUE THREAD

**Issue by arunasindhu19** [a85/POSTMan-Chrome-Extension#8747]
**Describe the bug**
When I try to call a resource I am seeing "Error: Parse Error: Invalid header value char" in the Response.

**To Reproduce**
N/A

**Expected behavior**
I should be able to get a response back with 200 OK.

**Screenshots**

**App information (please complete the following information):**
 - Postman Version [e.g. 7.27.1]
 - OS: [e.g. Windows 64 bit]

Additional Information:
I am seeing this issue only with One QA Environment. And the rest of the environments it works fine without changing any values in the header. 

![Postman Issue](https://user-images.githubusercontent.com/55460251/86833743-eedf5680-c067-11ea-9e08-26ff45ea3d46.jpg)

**Comment by reedbapex** [comment]
I am seeing this error also - starting with 7.27.1 - Win 64-bit.

![FailedHeader7271](https://user-images.githubusercontent.com/67982595/86849192-4fc55980-c07d-11ea-8700-f671a60f668f.png)

It works with previous version of Postman - making calls to same environment.  I have tried 7.7.3 & 6.3.0 (auto upgraded to 6.7.? & still worked).  A co-worker had 7.27.0 & that worked too, but as soon as they  restarted & upgraded to 7.27.1 - it started failing.  7.27.1 is failing in 4 of our 6 environments.  The console is no help as it won't display the pure raw response - just that there was a header parse failure.  I cannot determine the failing character using CURL -i & looking for invalid data.

![WorkingHeader773](https://user-images.githubusercontent.com/67982595/86849238-610e6600-c07d-11ea-8546-b0f2ccc7fa04.png)

**Comment by rodrigodesp** [comment]
A lot of people are having this problem where I work and here It´s blocked download old versions :S

**Comment by arunasindhu19** [comment]
@reedbapex Thank You for the quick response. We have been breaking our heads over what was wrong. We went back to the previous versions and it works. 7.18.0 and 7.20.1 versions are currently working.

**Comment by arunasindhu19** [comment]
@rodrigodesp That is frustrating. I hope your company does something about that :D

**Comment by SimonCigoj** [comment]
we have the same problem any news on this?

**Comment by sharedprophet** [comment]
I believe the issue is this: https://github.com/nodejs/node/issues/27711

TL;DR new node response parser is very strict with regard to header validity and many services inject invalid headers intentionally, leading to these issues. Later Postman versions must have come with an upgraded underlying node version.

What we need from Postman is the ability to set the "insecure" override option on the http parser.

**Comment by mohitranka** [comment]
@arunasindhu19 and others, Can you share the reproduction steps to the issue?

**Comment by sharedprophet** [comment]
1. Set up a server that will respond to a request with a header containing a newline, or other invalid character.
2. Send a request to it via Postman.

**Comment by viananicolas** [comment]
Creating a `NODE_OPTIONS `environment variable with the value  ` --http-parser=legacy` seems to have temporarily resolved this issue for me.

**Comment by reedbapex** [comment]
The NODE_OPTIONS environment variable from viananicola temporary resolution works for me too.

EDIT - the NODE_OPTIONS environment variable set system wide broke MS Teams - it doesn't like having the --http-parser NODE_OPTIONS set.  I had to set the environment variable when launching PostMan only.

**Comment by kkapell** [comment]
Can someone provide a little more details as to how to use this variable? Where to set it or how to reference it?

**Comment by reedbapex** [comment]
It will depend on you operating system & recommend the following instructions.

Win10 - https://www.architectryan.com/2018/08/31/how-to-change-environment-variables-on-windows-10/
Mac - https://medium.com/@himanshuagarwal1395/setting-up-environment-variables-in-macos-sierra-f5978369b255
Linux - https://www.serverlab.ca/tutorials/linux/administration-linux/how-to-set-environment-variables-in-linux/

For win10 - to launch with environment variable just for Postman, I modified the shortcut properties used to launch...

C:\Windows\System32\cmd.exe /c "SET NODE_OPTIONS=--http-parser=legacy&C:\Users\<user>\AppData\Local\Postman\Postman.exe"
I've also updated the 'Run' option to minimized.

**Comment by mdotson** [comment]
> The NODE_OPTIONS environment variable from viananicola temporary resolution works for me too.
> 
> EDIT - the NODE_OPTIONS environment variable set system wide broke MS Teams - it doesn't like having the --http-parser NODE_OPTIONS set. I had to set the environment variable when launching PostMan only.

This worked for me. If you set this environment variable in your shell and then launch the app from that same shell, the issue is resolved. For example on mac:

This in your `~/.bash_profile` or equivalent:
` export NODE_OPTIONS="--http-parser=legacy"`

Then you can launch postman like this (change path to wherever your postman is installed):
`open /Applications/Postman.app`

**Comment by mccannt** [comment]
Just wanted to drop a quick update here for everyone. We have this currently in our backlog to be resolved, thankfully it looks like we have the workaround listed above for those that are impacted by this. 
We will let you know when we have any updates.

**Comment by Sasi-21** [comment]
Hi All, I am also getting parse error. Is there any updates on this issue. As lots of people facing same issue. 

![image](https://user-images.githubusercontent.com/63800698/100188888-32e8d000-2f11-11eb-9077-cf45814d6b5a.png)

**Comment by reedbapex** [comment]
There are relatively stable work-arounds listed above in the Jul-29 - Sept-16 comments.  The Postman team has not addressed it as of 7.36.

**Comment by VaseemDeveloper** [comment]
Is there any update? If this issue is resolved for someone?
I have set the environment variable and restarted the Postman - Version 7.36.0 (7.36.0)
But this approach is not working.
printenv
NODE_OPTIONS=--http-parser=legacy

**Comment by uniquemag** [comment]
is there solution for this issue?  Also encounter this on postman version 7.36.0
function was working 2 days ago, but no clue to see what's the problem now.

**Comment by Sasi-21** [comment]
Hi All, setting environment variable approach working fine for me.
Just add following in the environment variables,

NODE_OPTIONS = --http-parser=legacy


![image](https://user-images.githubusercontent.com/63800698/101353806-3c334e80-38ba-11eb-857a-4cb7a9303729.png)

**Comment by actongorton** [comment]
This issue came up. Turned out that the incoming response header from the api endpoint contained conflicting HTTP version declarations: `HTTP/1.1 200 OK` and `HTTP/1.0 200`. In my case, a custom response was being written into the initial response payload, and then a second version declaration was being written into the payload by the proxy. Removing the initial declaration and allowing the proxy to have its way solved the issue for me.

I was able to identify the problem by using `curl` with the verbose (`-v`) output to view and compare the responding headers against the expected headers.

**Comment by ramen-mukherjee** [comment]
I upgraded to Node v15.5.0 x64 today. With Postman v7.36.1 it started giving same issue.
I am on Windows 10 x64 and tried to set NODE_OPTIONS = --http-parser=legacy.
After setting this Postman doesn't start up.
Is there any other way to bypass this?

**Comment by reedbapex** [comment]
My understanding is that above workaround doesn't work after Node 12.   It's a node parser caused issue with some of the anti-DOS software solutions - or if the API is returning bad data in the headers.

**Comment by ramen-mukherjee** [comment]
@reedbapex : Is Postman Dependent on NodeJS ?

**Comment by iyosayi** [comment]
Is there a fix for this issue now? I tried setting the `NODE_OPTIONS` env in Windows 10 but didn't work. In my case, I was able to make a successful `GET` request. When I tried it the second time, I started getting this same exact error, is that the responses are being cached somewhere?

**Comment by dfaliush** [comment]
the same issue: Error: Parse Error: Invalid header value char

**Comment by BryanVision** [comment]
I can't believe this is still an issue and postman is still broken for me and that I can't find an answer.

**Comment by tplunket** [comment]
I'm also running into this, in my case the response header is indeed invalid:
```
HTTP/1.1 200 OK
Content-Length: -1
Transfer-Encoding: chunked
```
I understand that the server code is in violation of the standard for a couple of reasons and I can't fix that right now but the "parse error" messaging is useless without a "show raw response" option. (The fact that I have to use yet another tool to actually get these headers devalues Postman.) It'd be super extra awesome if it could indicate where the parse error is specifically but maybe the library doesn't provide enough context. Also it would be nice if it would carry on the best it could even in the face of errors like this, e.g. my client code does "if content_length > 0" before doing anything with any possible body.

A development tool like this should definitely follow the "conservative in what you do, lenient in what you accept" maxim even if that isn't desirable these days in consumer-facing software. IMHO. ;) Surface errors but be able to avoid the abort on error so that I can focus on my actual tasks.

**Comment by cyopen1987** [comment]
terrible mistake to upgrade, same issue, @viananicolas 's solution works .

**Comment by mhartzell-incomm** [comment]
How in the WORLD has this not been addressed in over a year?  What Postman replacement is out there with which I can invoke a basic GET request?

**Comment by cyopen1987** [comment]
> How in the WORLD has this not been addressed in over a year? What Postman replacement is out there with which I can invoke a basic GET request?

@mhartzell-incomm  as a chinese develper , I use postwoman instead, you can try , https://chrome.google.com/webstore/detail/postwoman-http%E6%8E%A5%E5%8F%A3%E8%B0%83%E8%AF%95%E6%8F%92%E4%BB%B6/ieoejemkppmjcdfbnfphhpbfmallhfnc
 it 's a chrome plugin

**Comment by stdunbar** [comment]
v9.2.0 on Ubuntu with or without the NODE_OPTIONS setting is currently breaking the same way.  For this example Postman is unusable and there is nothing output from Postman to indicate which header Postman considers invalid.

**Comment by RaymondMwaura** [comment]
In my case, saving newly entered environment variables then restarting Postman did the trick. Seems the "Save" feature has a bug.

**Comment by cattoa** [comment]
It seems that 9.3.1 on Windows is also not working with or without a NODE_OPTIONS settings in the environment variables

**Comment by PRRB** [comment]
I'm on Postman v9.5.0 and I'm running into the same issue since yesterday.
- Only affects our dev environment, all others qa/uat etc. are fine.
- It affects all requests to this domain.
- I can make requests to this domain with Chrome without any problem.
- NODE_OPTIONS environment variable workaround suggested above does not work.

It's crazy that this is still an issue which others were reporting over a year ago. I cannot even inspect the faulty headers in Postman, the only information is "There seems to be an invalid character in the response header value".

**Comment by ParrySMS** [comment]
Postman v9.7.1  meet the same issue. Finally I fix it by other way.

Attention: Do not set the NODE_OPTIONS in environment variable for system or postman.

If it not works for you, please remove it. Otherwise, it may cause your normal request to meet "Parse Error: There seems to be an invalid character in the response header value".

I fix it by disable some postman config. I am not sure which one have effect(maybe proxy). Here is my setting.

I turn on the "Send no-cache header", turn off the "SSL certificate verification" and off "Send Postman Token header" in the General tag.

I turn off the "CA  certificates" in Certificates tag.

I click off the "Use the system proxy" in Proxy tag.


![截屏2021-12-31 下午2 11 35](https://user-images.githubusercontent.com/27364780/147806898-50a0b258-f3b1-49b7-892e-52042f3fe4bc.png)

![截屏2021-12-31 下午2 16 31](https://user-images.githubusercontent.com/27364780/147806907-1ae9ee48-5e00-441a-adc0-d6d23f769d44.png)

![截屏2021-12-31 下午2 16 37](https://user-images.githubusercontent.com/27364780/147806916-08da153d-c852-488c-ac30-59805d5c0c15.png)

**Comment by ParrySMS** [comment]
> I'm on Postman v9.5.0 and I'm running into the same issue since yesterday.
> 
> * Only affects our dev environment, all others qa/uat etc. are fine.
> * It affects all requests to this domain.
> * I can make requests to this domain with Chrome without any problem.
> * NODE_OPTIONS environment variable workaround suggested above does not work.
> 
> It's crazy that this is still an issue which others were reporting over a year ago. I cannot even inspect the faulty headers in Postman, the only information is "There seems to be an invalid character in the response header value".

Please see my new comment. It maybe helpful.

**Comment by zfeed** [comment]
You can use `insecureHTTPParser` option to bypass it

See [here](https://nodejs.org/api/http.html#httprequesturl-options-callback)

**Comment by monzonj** [comment]
Alternative to postman: https://hoppscotch.io/

**Comment by shamseerahammedm** [comment]
this issue was reported at **7.27.1**  now its **9.14.11** and still facing the same issue

**Comment by SimonCigoj** [comment]
same for me ... stil the same problem, I have not found a workaround

**Comment by ErnandesAJr** [comment]
same for me ... stil the same problem, I have not found a workaround

**Comment by Bartmax** [comment]
same here

**Comment by unclosable** [comment]
so which header is invalid exactly

**Comment by shanks25** [comment]
do we any update

**Comment by matisidler** [comment]
waiting for a solution :|

**Comment by MichaelHallik** [comment]
And here as well. :-(

**Comment by ghost** [comment]
When using Imperva, the Issue was caused by a server classification cookie coontaining invalid characters for some HTTP parsers. Disabling this cookie solved the issue for me.

**Comment by bigfoot31** [comment]
This definitely is a problem & needs to be fixed. Otherwise Postman is proving useless for me. Here are the headers for which the above error is occurring. 
![Screenshot 2022-10-13 at 22 38 48](https://user-images.githubusercontent.com/14835245/195628112-009173c2-5c1a-4192-aee0-fa62ff069d4a.png)

**Comment by wenlong139** [comment]
还没解决吗 真服了

**Comment by wenlong139** [comment]
I meet this issue. But I think this issue is don'n releated postman. 
I find when I curl  an URL which response header has a key "C2vydmljzs1tzxnolw1hbmfnzxi=". 
The head key contains a character '='  It is a special character  and it's why postman analyze response invalid
So  the reason may be your interface response a invalid response header key

**Comment by giridharvc7** [comment]
Hello, we've fixed this issue with v10.6.x. Please update your app for these changes.
Thank you for your patience, please let us know if you face any issues.

**Comment by imhamzaaslam** [comment]
You are up to date!
Postman v10.6.7 is the latest version.
Last checked: December 23, 2022 at 6:00 PM

and still facing same issue. @giridharvc7

**Comment by giridharvc7** [comment]
@imhamzaaslam Can you share some more details on this?

**Comment by ankur586** [comment]
I have updated to v10.10.6 but still facing the same issue.

**Comment by ankur586** [comment]
> 

I have updated to v10.10.6 but still facing the same issue.

**Comment by ocierd** [comment]
I have the same issue; i can't resolve in postman, but when send the request by CURL, i see the error in the **Strict-Transport_Security** header. Any idea? Thanks!

![image](https://user-images.githubusercontent.com/38734951/236586740-b6581241-c04b-491b-b59a-c96101e57521.png)

**Comment by ismdcf** [comment]
Hi Team, 

Any update on this 
My Reponse header has a weird response to it 
<img width="515" alt="Screenshot 2024-06-03 at 10 55 42 AM" src="https://github.com/postmanlabs/postman-app-support/assets/16378404/58f8b7db-a07d-42bb-be4d-b6376c319e71">

Is there any possibility to disable the response header parsing on postman

My postman version is 
Version 11.1.14
UI Version: 11.1.14-ui-240521-0853
Desktop Platform Version: 11.1.14 (11.1.14)
